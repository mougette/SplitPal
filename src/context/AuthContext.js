import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
