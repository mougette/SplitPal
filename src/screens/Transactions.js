import React, {useContext, useState, useEffect} from 'react';
import {View, Text, StyleSheet, Image, SafeAreaView, FlatList } from 'react-native';
import {Button, Input, Icon} from 'react-native-elements';
import {Context as AuthContext} from '../context/AuthContext';
import SplitPalLogoComponent from '../components/SplitPalLogoComponent';
import Entry from "../components/EntryComponent";
import {Get} from '../components/RestGet';

const Transactions = ({route, navigation}) => {
  const {itemEmail} = route.params;
  const {state, signout} = useContext(AuthContext);
  const [DATA, setDATA] = useState("");

  useEffect(() => {
  const unsubscribe = navigation.addListener('focus', () => {
  Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/transaction","?userEmail="+state.email+"&friendEmail="+itemEmail)
      .then(response => setDATA(JSON.parse(response)));
  });
  return unsubscribe
  },[navigation]);

  const renderItem = ( {item,index} ) => {
    return (
        <Entry
            image='https://reactnative.dev/img/tiny_logo.png'
            name={item.description}
            balance={item.amount.toString()}
            onPress = {() => console.log("TODO: Entry Pressed on Transactions Screen")}
        ></Entry>
    );
  };

  
  return (
    <View style={styles.master}>
      <SplitPalLogoComponent />

      <Image source ={{uri: 'https://reactnative.dev/img/tiny_logo.png'}}
             style={{width:60, height:60,borderRadius:30}} />

      <Text style={styles.header}>Your Transactions With, {itemEmail}:</Text>

      <SafeAreaView style={styles.containerTop}>
        <FlatList
            data={DATA == "" ? DATA : DATA.splice(1)}
            renderItem={renderItem}
            keyExtractor = {(item) => item.transactionID.toString()}
            extraData={DATA}
        />
      </SafeAreaView>

    </View>
  );
};

const styles = StyleSheet.create({
  master: {
    flex: 1,
    backgroundColor: '#87a1b6',
    alignItems: 'center',
  },
  header: {
    fontSize: 20,
    marginBottom: 8,
    marginTop: 20,
  },
});

export default Transactions;
 