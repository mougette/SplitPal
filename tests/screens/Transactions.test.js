import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import Transactions from '../../src/screens/Transactions'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve("[{\"amount\": -40.0, \"userOwed\": \"e@something.com\", \"userOwes\": \"shogenson@wisc.edu\", \"receiptID\": 29, \"description\": \"Lunch\", \"transactionID\": 54}, {\"amount\": -10.0, \"userOwed\": \"e@something.com\", \"userOwes\": \"shogenson@wisc.edu\", \"receiptID\": 40, \"description\": \"Cherse\", \"transactionID\": 69}]"),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Profile tests', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Profile', async () => {
        const {getByText} = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="Transactions" component={Transactions} initialParams ={{itemEmail : "e@something.com"}} />
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
        await waitFor(() => {
        expect(1).toBe(1);
        })})


});