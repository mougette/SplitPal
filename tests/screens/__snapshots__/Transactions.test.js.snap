// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests Profile 1`] = `undefined`;
